pkg/util/file.go

```diff
@@ -261,28 +264,50 @@ func StreamDataToFile(r io.Reader, fileName string, preallocate bool) (int64, in
 
 type zeroWriterFunc func(*os.File, int64, int64) error
 
+func zeroWriterWithFallback(zwf zeroWriterFunc) func(dst *os.File, start, length int64) (int64, error) {
+	return func(dst *os.File, start, length int64) (int64, error) {
+		err := zwf(dst, start, length)
+		if err != nil {
```

Comment:
should we be more precise in here? something like https://github.com/containers/storage/blob/10981abafe3e3ece4e0cc872761f7624cd1bef0e/drivers/copy/copy_linux.go#L111

Reply:
Figured best to do exactly what we are doing here: https://github.com/kubevirt/containerized-data-importer/blob/main/pkg/importer/imageio-datasource.go#L309. I don't think we can count on all provisioners returning the "right" error

I doubt that transient errors happen frequently and since "retrying over transient errors" is starting the entire clone operation over from the beginning I think this solution is fine